src/main/minio.js

```diff
@@ -2123,6 +2123,62 @@ export class Client {
     return listener
   }
 
+  getBucketVersioning(bucketName,cb) {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
+    }
+    if (!isFunction(cb)) {
+      throw new errors.InvalidArgumentError('callback should be of type "function"')
+    }
+    var method = 'GET'
+    var query = "versioning"
+
+    this.makeRequest({method, bucketName, query}, '', 200, '', true, (e, response) => {
+      if (e) return cb(e)
+
+      let versionConfig = Buffer.from('')
+      pipesetup(response, transformers.bucketVersioningTransformer())
+        .on('data', data => {
+          versionConfig = data
+        })
+        .on('error', cb)
+        .on('end', () => {
+          cb(null, versionConfig)
+        })
+    })
+  }
+
+  setBucketVersioning(bucketName,versionConfig, cb) {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
+    }
+    if(!Object.keys(versionConfig).length){
+      throw new errors.InvalidArgumentError('versionConfig should be of type "object"')
+    }
+    if (!isFunction(cb)) {
+      throw new TypeError('callback should be of type "function"')
+    }
+
+    var method = 'PUT'
+    var query = "versioning"
+    var builder = new xml2js.Builder({rootName:'VersioningConfiguration', renderOpts:{'pretty':false}, headless:true})
+    var payload = builder.buildObject(versionConfig)
+
+    var versionInfo = Buffer.from('')
+    this.makeRequest({method, bucketName, query}, payload, 200, '', true, (e, response) => {
+      if (e) return cb(e)
+
+      pipesetup(response, transformers.bucketVersioningTransformer())
+        .on('data', data => {
+          versionInfo = data
+        })
+        .on('error', cb)
+        .on('end', () => {
+          cb(null, versionInfo.toString())
+        })
+    })
```

Comment:
Sorry for not noticing this before.
This part can be removed. We don't return anything other than `error` for `setBucketVersioning`, or for `PUT` method.

```suggestion
    this.makeRequest({method, bucketName, query}, payload, 200, '', true, cb)
```
If you'd like to see examples, please check the other methods, `setBucketNotification`, or `setBucketPolicy`.
Please don't forget to also touch the corresponding references in the callback function and the returned value, `result`, in the docs and the example.